p2p/host/basic/basic_host.go

```diff
@@ -224,6 +243,19 @@ func NewHost(ctx context.Context, net network.Network, opts *HostOpts) (*BasicHo
 	return h, nil
 }
 
+func getLocalIPAddrFor(ip net.IP) (net.IP, error) {
+	r, err := netroute.New()
```

Comment:
should be cached. as used this'll build up the full OS routing table twice. ideally make one netroute object for both the v4 and v6 query.

Reply:
This is done.